diff --git a/dist/rrweb.js b/dist/rrweb.js
index 1db2d08b2fe9a73521001c3aa88b1a64d1b3e71a..5a3ea8410b0f1dfa7248b9cb9752c04eea7b3ad3 100644
--- a/dist/rrweb.js
+++ b/dist/rrweb.js
@@ -85,6 +85,9 @@ function isCSSImportRule(rule) {
 function isCSSStyleRule(rule) {
   return "selectorText" in rule;
 }
+function findStylesheet(doc, href) {
+  return Array.from(doc.styleSheets).find((s) => s.href === href);
+}
 class Mirror {
   constructor() {
     __publicField$1(this, "idNodeMap", /* @__PURE__ */ new Map());
@@ -661,9 +664,14 @@ function serializeElementNode(n2, options) {
     }
   }
   if (tagName === "link" && inlineStylesheet) {
-    const stylesheet = Array.from(doc.styleSheets).find((s2) => {
-      return s2.href === n2.href;
-    });
+    const href = n2.href;
+    let stylesheet = findStylesheet(doc, href);
+    if (!stylesheet && href.includes('.css')) {
+      const rootDomain = window.location.origin
+      const stylesheetPath = href.replace(window.location.href, '')
+      const potentialStylesheetHref = rootDomain + '/' + stylesheetPath
+      stylesheet = findStylesheet(doc, potentialStylesheetHref);
+    }
     let cssText = null;
     if (stylesheet) {
       cssText = stringifyStylesheet(stylesheet);
@@ -5016,11 +5024,19 @@ class CanvasManager {
     let rafId;
     const getCanvas = () => {
       const matchedCanvas = [];
-      win.document.querySelectorAll("canvas").forEach((canvas) => {
-        if (!isBlocked(canvas, blockClass, blockSelector, true)) {
-          matchedCanvas.push(canvas);
-        }
-      });
+      const searchCanvas = (root) => {
+        root.querySelectorAll("canvas").forEach((canvas) => {
+          if (!isBlocked(canvas, blockClass, blockSelector, true)) {
+            matchedCanvas.push(canvas);
+          }
+        });
+        root.querySelectorAll("*").forEach((elem) => {
+          if (elem.shadowRoot) {
+            searchCanvas(elem.shadowRoot);
+          }
+        });
+      };
+      searchCanvas(win.document);
       return matchedCanvas;
     };
     const takeCanvasSnapshots = (timestamp) => {
@@ -5041,13 +5057,20 @@ class CanvasManager {
             context.clear(context.COLOR_BUFFER_BIT);
           }
         }
-        const bitmap = await createImageBitmap(canvas);
+        // createImageBitmap throws if resizing to 0
+        // Fallback to intrinsic size if canvas has not yet rendered
+        const width = canvas.clientWidth || canvas.width;
+        const height = canvas.clientHeight || canvas.height;
+        const bitmap = await createImageBitmap(canvas, {
+          resizeWidth: width,
+          resizeHeight: height
+        });
         worker.postMessage(
           {
             id,
             bitmap,
-            width: canvas.width,
-            height: canvas.height,
+            width: width,
+            height: height,
             dataURLOptions: options.dataURLOptions
           },
           [bitmap]
@@ -9805,6 +9828,7 @@ class Replayer {
 }
 const { addCustomEvent } = record;
 const { freezePage } = record;
+const version = "2.0.0-alpha.16"
 export {
   EventType,
   IncrementalSource,
@@ -9816,6 +9840,7 @@ export {
   freezePage,
   _mirror as mirror,
   record,
-  utils
+  utils,
+  version,
 };
 //# sourceMappingURL=rrweb.js.map
